### DISCUSSION TITLE
Nullable ref Value property

### DISCUSSION THREAD

**Discussion by emceelovin** [dotnet/runtime#73117] [General] [answered]
# FEATURE_NAME

* [x] Proposed
* [ ] Prototype: Not Started
* [ ] Implementation: Not Started
* [ ] Specification: Not Started

## Summary
To avoid CS1612's, unnecessary copies, and because it just makes sense, Nullable<T>.Value should be updated to return by reference.

## Detailed design
```cs
public struct NestedValueType
{
    public string StringProp { get; set; }
}

public struct Nullable<T>
{
    // public ref T Value => ref _value
    public T Value => _value;

    public ref T ByRefValue => ref _value;

    T _value;
}

public void Serialize(ref Nullable<NestedValueType> obj)
{
    /*
     * CS1612. StringProperty is returned as a copy as it's the return value of the internal getter and not a variable.
     */
    obj.Value.StringProp = reader.ReadString();

    /*
     * Bingo. Zero copies of any of the value type's anywhere. The desirable should-be outcome. 
     */
    obj.ByRefValue.StringProp = string.Empty;

    /*
     * This shouldn't break anything
     * either. If you were to try and assign the ref read property to a non reference, it would just be copied like the typical
     * non by ref property
     */
    NestedValueType copy = obj.ByRefValue;
}
```

## Drawbacks
You tell me?

## Alternatives
```cs
// I emitted the IL version of this with DynamicMethod and ILGenerator
public ref T GetNullableValueByRef<T>(ref Nullable<T> nullable)
{
    return ref nullable._value;
}
```

**Comment by CyrusNajmabadi** [comment]
Moving to runtime.  This is not a language decision.

**Comment by PathogenDavid** [comment, marked as answer]
[`Nullable.GetValueRefOrDefaultRef<T>`](https://docs.microsoft.com/en-us/dotnet/api/system.nullable.getvaluerefordefaultref?view=net-7.0) has been accepted to .NET 7 and is already available in .NET 7 previews.

(See PR https://github.com/dotnet/runtime/pull/64677 and API proposal https://github.com/dotnet/runtime/issues/1534)

**Reply by emceelovin** [reply]
Thank you, David, and I figured I should've put this in runtime. All of that is great to hear!
